spiketoolkit/validation/qualitymetrics.py

```diff
@@ -25,28 +38,85 @@ def compute_unit_SNR(recording, sorting, unit_ids=None, save_as_property=True):
     '''
     if unit_ids is None:
         unit_ids = sorting.get_unit_ids()
-    channel_noise_levels = _compute_channel_noise_levels(recording=recording)
-    templates = st.postprocessing.postprocessing_tools.get_unit_template(recording, sorting, unit_ids=unit_ids)
+    if filter:
+        recording_f = st.preprocessing.bandpass_filter(recording=recording, freq_min=freq_min, freq_max=freq_max)
+    else:
+        recording_f = recording
+    channel_noise_levels = _compute_channel_noise_levels(recording=recording_f, mode=mode, seconds=seconds)
+    templates = st.postprocessing.get_unit_template(recording, sorting, unit_ids=unit_ids,
+                                                    max_num_waveforms=max_num_waveforms,
+                                                    mode='median')
+    max_channels = st.postprocessing.get_unit_max_channel(recording, sorting, unit_ids=unit_ids,
+                                                          max_num_waveforms=max_num_waveforms,
+                                                          mode='median')
     snr_list = []
     for i, unit_id in enumerate(sorting.get_unit_ids()):
-        snr = _compute_template_SNR(templates[i], channel_noise_levels)
+        max_channel_idx = recording.get_channel_ids().index(max_channels[i])
+        snr = _compute_template_SNR(templates[i], channel_noise_levels, max_channel_idx)
         if save_as_property:
             sorting.set_unit_property(unit_id, 'snr', snr)
         snr_list.append(snr)
     return snr_list
 
 
-def _compute_template_SNR(template, channel_noise_levels):
-    channel_snrs = []
-    for ch in range(template.shape[0]):
-        channel_snrs.append((np.max(template[ch, :]) - np.min(template[ch, :])) / channel_noise_levels[ch])
-    return np.max(channel_snrs)
+def _compute_template_SNR(template, channel_noise_levels, max_channel_idx):
+    '''
+    Computes SNR on the channel with largest amplitude
+
+    Parameters
+    ----------
+    template: np.array
+        Template (n_elec, n_timepoints)
+    channel_noise_levels: list
+        Noise levels for the different channels
+    max_channel_idx: int
+        Index of channel with largest templaye
+
+    Returns
+    -------
+    snr: float
+        Signal-to-noise ratio for the template
+    '''
+    snr = np.max(np.abs(np.min(template[max_channel_idx]))) / channel_noise_levels[max_channel_idx]
```

Comment:
In case of positive spike I would do
max( np.max(np.abs(np.min(template[max_channel_idx]))), np.max(template[max_channel_idx]) ) / ...

Reply:
I need to change the get_unit_max_channel as well, because it uses the same.

maybe just a np.max(np.abs(template[max_channel_idx]))??